test/bigtable/row.js

```diff
@@ -0,0 +1,785 @@
+    });
+
+    it('should return an error to the callback', function(done) {
+      var err = new Error('err');
+      var response = {};
+
+      row.parent.mutate = function(entry, callback) {
+        callback(err, response);
+      };
+
+      row.create(function(err_, row, apiResponse) {
+        assert.strictEqual(err, err_);
+        assert.strictEqual(row, null);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+
+    it('should return the Row instance', function(done) {
+      var response = {};
+
+      row.parent.mutate = function(entry, callback) {
+        callback(null, response);
+      };
+
+      row.create(function(err, row_, apiResponse) {
+        assert.ifError(err);
+        assert.strictEqual(row, row_);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+  });
+
+  describe('createRules', function() {
+    var rules = [{
+      column: 'a:b',
+      append: 'c',
+      increment: 1
+    }];
+
+    it('should read/modify/write rules', function(done) {
+      row.request = function(grpcOpts, reqOpts) {
+        assert.deepEqual(grpcOpts, {
+          service: 'BigtableService',
+          method: 'readModifyWriteRow'
+        });
+
+        assert.strictEqual(reqOpts.tableName, TABLE.id);
+        assert.strictEqual(reqOpts.rowKey, ROW_ID);
+
+        assert.deepEqual(reqOpts.rules, [{
+          familyName: 'a',
+          columnQualifier: 'b',
+          appendValue: 'c',
+          incrementAmount: 1
+        }]);
+
+        var spy = FakeMutation.convertToBytes;
+
+        assert.strictEqual(spy.getCall(0).args[0], 'b');
+        assert.strictEqual(spy.getCall(1).args[0], 'c');
+        assert.strictEqual(spy.getCall(2).args[0], ROW_ID);
+        done();
+      };
+
+      row.createRules(rules, assert.ifError);
+    });
+
+    it('should return an error to the callback', function(done) {
+      var err = new Error('err');
+      var response = {};
+
+      row.request = function(g, r, callback) {
+        callback(err, response);
+      };
+
+      row.createRules(rules, function(err_, apiResponse) {
+        assert.strictEqual(err, err_);
+        assert.strictEqual(response, apiResponse);
+        done();
+      });
+    });
+  });
+
+  describe('filter', function() {
+    var mutations = [{
+      method: 'insert',
+      data: {
+        a: 'a'
+      }
+    }];
+
+    beforeEach(function() {
+      FakeMutation.parse.reset();
+    });
+
+    it('should provide the proper request options', function(done) {
+      var filter = {
+        column: 'a'
+      };
+
+      row.request = function(grpcOpts, reqOpts) {
+        assert.deepEqual(grpcOpts, {
+          service: 'BigtableService',
+          method: 'checkAndMutateRow'
+        });
+
+        assert.strictEqual(reqOpts.tableName, TABLE.id);
+        assert.strictEqual(reqOpts.rowKey, ROW_ID);
+        assert.deepEqual(reqOpts.predicateFilter, filter);
```

Comment:
Need to be sure `reqOpts.predicateFilter` is the result of calling `Filter.parse(filter)`.